File: app/src/main/java/com/app/syspoint/db/dao/TaskDao.java
package com.app.syspoint.db.dao;


import com.app.syspoint.db.bean.TaskBean;
import com.app.syspoint.db.bean.TaskBeanDao;

import java.util.List;

public class TaskDao extends Dao{

    public TaskDao() {
        super("TaskBean");
    }


    public final TaskBean getTask(String date){
        final List<TaskBean> taskBeans = dao.queryBuilder()
                .where(TaskBeanDao.Properties.Date.eq(date  ))
                .list();
        return taskBeans.size()> 0? taskBeans.get(0) : null;
    }

}

File: app/src/main/java/com/app/syspoint/repository/objectBox/dao/TaskDao.kt
package com.app.syspoint.repository.objectBox.dao

import com.app.syspoint.repository.objectBox.entities.TaskBox
import com.app.syspoint.repository.objectBox.entities.TaskBox_
import io.objectbox.query.QueryBuilder

class TaskDao: AbstractDao<TaskBox>() {

    fun clear() {
        abstractBox<TaskBox>().removeAll()
    }

    fun getTask(date: String?): TaskBox? {
        val query = abstractBox<TaskBox>().query()
            .equal(TaskBox_.date, date, QueryBuilder.StringOrder.CASE_INSENSITIVE)
            .build()
        val results = query.find()
        query.close()

        return if (results.isNotEmpty()) results[0] else null
    }
}